hail/python/test/hail/test_ir.py

```diff
@@ -570,3 +572,52 @@ def test_refs(self):
                     ' (bar (GetField idx (Ref row)))))'
         )
         assert expected == CSERenderer()(x)
+
+
+def _assert_encoding_roundtrip(value):
+    lit = hl.literal(value)
+    round_trip = lit.dtype._from_encoding(lit.dtype._to_encoding(value))
+    if isinstance(value, np.ndarray):
+        assert_array_equal(round_trip, value)
+    else:
+        assert round_trip == value
+
+
+@pytest.mark.parametrize(
+    'value',
+    [
+        1,
+        5.0,
+        "foo",
+        [1, 2, 3, 4],
+        (5, 6, 7, 8),
+        {"foo", "bar", "baz"},
+        {"a": 1, "b": 2},
+        np.array([[1, 2], [3, 4], [5, 6]]),
+        np.array([[1, 2], [3, 4], [5, 6]]).T,
```

Comment:
Pursuant to your comment above can we add:

1. Test for 0d np.array
2. Test for 1d np.array
3. Test for 2d np.array
4. Test for 4d np.array
5. Transposition of all of the above